File: apps/pos-flutter/lib/features/audit/audit_service.dart
import 'dart:convert';

import '../../core/local_db.dart';
import '../../core/time_utils.dart';

class AuditService {
  Future<void> log({
    required String action,
    required String actor,
    required String roleName,
    required Map<String, dynamic> payload,
  }) async {
    final db = await LocalDb.open();
    await db.insert('audit_logs', {
      'action': action,
      'actor': actor,
      'role_name': roleName,
      'payload_json': jsonEncode(payload),
      'created_at': nowIso(),
    });
  }
}

File: backend/app/Services/AuditService.php
<?php

namespace App\Services;

use Illuminate\Support\Facades\DB;

class AuditService
{
    public function log(?int $tenantId, ?int $outletId, ?string $actor, string $eventType, ?string $entityType = null, ?string $entityId = null, array $payload = []): void
    {
        DB::table('audit_logs')->insert([
            'tenant_id' => $tenantId,
            'outlet_id' => $outletId,
            'actor' => $actor,
            'event_type' => $eventType,
            'entity_type' => $entityType,
            'entity_id' => $entityId,
            'payload_json' => json_encode($payload, JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES),
            'created_at' => now(),
            'updated_at' => now(),
        ]);
    }
}

